### DISCUSSION TITLE
ESP32 unable to erase flash

### DISCUSSION THREAD

**Discussion by gregfox151** [micropython/micropython#13025] [ESP32] [answered]
Hello all, I would like some assistance in flashing micropython to an esp32 
I downloaded micropython from https://micropython.org/ 
I understand one must erase before flashing in micropython.
My setup is a Win 11 Pro Version 10.0.22621 Build 22621 (installed WSL and Ubuntu)
Board on com17.
My board is an esp32 Devkit V1 with esp32 on board  WROOM.
I want to use [v1.21.0 (2023-10-05) .bin](https://micropython.org/resources/firmware/ESP32_GENERIC-20231005-v1.21.0.bin) after I’m able to erase my esp32.
_____________________________________________________________
I use the following code to erase the esp32 while holding the boot key:
C:\M_Python> esptool.py --chip esp32  erase_flash
I also tried:
                          esptool.py --chip esp32  --com17 erase_flash
AND I GOT:
[main 2023-11-19T18:56:05.414Z] update#setState idle
[main 2023-11-19T18:56:35.416Z] update#setState checking for updates
[main 2023-11-19T18:56:35.516Z] update#setState idle
With this Visual Studio opens and ...
PS C:\Users\greg\Documents\MicroPython> & C:/Python312/python.exe c:/M_Python/esptool.py
Could not find platform independent libraries <prefix>
Traceback (most recent call last):
  File "c:\M_Python\esptool.py", line 34, in <module>
    import esptool
  File "c:\M_Python\esptool\__init__.py", line 41, in <module>
    from esptool.bin_image import intel_hex_to_bin
  File "c:\M_Python\esptool\bin_image.py", line 16, in <module>
    from intelhex import IntelHex
ModuleNotFoundError: No module named 'intelhex'
PS C:\Users\greg\Documents\MicroPython> 

Does anyone have a clue as to what I’m doing wrong?
Thank you.

**Comment by kjm1102** [comment]
The Thonny IDE gets some bad press on this site but it mostly does a no tears, menu driven job of flashing an esp32 with the latest upython. Just sayin'.

**Comment by MrTinkerman** [comment, marked as answer]
I got this same message:
"ModuleNotFoundError: No module named 'intelhex'"

What solved it for me was: pip install intelhex
[(https://pypi.org/project/intelhex/)](https://pypi.org/project/intelhex/)

**Reply by gregfox151** [reply]
Thanks, that did it. I did the following after downloading ESP32_GENERIC-20231005-v1.21.0.bin:

pip install esptool

esptool --port com17 erase_flash

esptool --chip esp32 --port com17 write_flash -z 0x1000 ESP32_GENERIC-20231005-v1.21.0.bin

**Comment by gregfox151** [comment]
Thanks, ill give it a shot.

**Comment by DavesCodeMusings** [comment]
This works for me in Windows 11 with the same board...

1. Download latest python windows 64bit from https://www.python.org/
2. Install for all users accepting the defaults in the installer wizard.
3. From a powershell prompt: `py -m pip install esptool`
4. Then run esptool with: `py -m esptool --chip auto --port ${port} write_flash --erase-all --flash_size=detect ${address} ${image}`

${port} is your COM port
${address} is 0x1000 for ESP32, 0x0 for ESP32-C3, and ESP32-S3.
${image} is your firmware file.

For example: `py -m esptool --chip auto --port COM17 write_flash --erase-all --flash_size=detect 0x1000 ESP32_GENERIC-20231005-v1.21.0.bin`

**Comment by gregfox151** [comment]
It works!  Thanks
